annotations/component-annotations/src/main/java/io/dekorate/component/handler/ComponentHandler.java

```diff
@@ -110,6 +124,20 @@ private void addVisitors(ComponentConfig config) {
     }
   }
 
+  private KubernetesConfig getKubernetesConfig() {
```

Comment:
We could possibly avoid the dependency to `kubernetes-annotations` by using BaseConfig instead of KubernetesConfig.

However, I don't expect

       configurators.get(BaseConfig.class)


to work. So maybe have Configurators internally iterate and check using `isAssignable` instead of just delegating to a map?

Reply:
The good thing is that you have already done that work :wink: 

So it should work OOTB

Reply:
But of course when I tried it failed....

Reply:
I have?

Reply:
Wow, just checked the implementation of `get`. I have completly forgoten about it.

Reply:
:)

That part seems to work. Something else if failing but I'm tracking it doen